data-prepper-api/src/main/java/org/opensearch/dataprepper/model/source/coordinator/SourceCoordinator.java

```diff
@@ -94,4 +96,14 @@ public interface SourceCoordinator<T> {
      * @since 2.2
      */
     void giveUpPartitions();
+
+
+    /**
+     * Should be called by the source after when acknowledgments are enabled to keep ownership of the partition for acknowledgmentTimeout amount of time
+     * before another instance of Data Prepper can pick it up for processing. Allows the source to acquire another partition immediately for processing
+     * @param partitionKey - the partition to update for ack timeout
+     * @param ackowledgmentTimeout - the amount of time that this partition can be completed by the acknowledgment callback before another instance of Data Prepper
+     *                             can pick it up for processing
+     */
+    void updatePartitionForAckWait(final String partitionKey, final Duration ackowledgmentTimeout);
```

Comment:
Let's use full words in interface names. `updatePartitionForAcknowledgementWait`.